agents/src/transcription.ts

```diff
@@ -118,12 +126,72 @@ export class BasicTranscriptionForwarder implements TranscriptionForwarder {
   }
 
   async close(interrupt: boolean): Promise<void> {
-    this.#isRunning = false;
+    this.isRunning = false;
 
     // Publish whatever we had as final
     if (!interrupt) {
-      this.currentCharacterIndex = this.#currentText.length;
+      this.currentCharacterIndex = this.currentText.length;
+    }
+    await this.publishTranscription(true);
+  }
+}
+
+export class ChatAndTranscriptionForwarder extends BasicTranscriptionForwarder {
+  protected type: TranscriptionType;
+  protected originalMessage?: ChatMessage;
+  protected chatMutex: Mutex;
+
+  constructor(
+    room: Room,
+    participantIdentity: string,
+    trackSid: string,
+    messageId: string,
+    transcriptionType: TranscriptionType,
+  ) {
+    super(room, participantIdentity, trackSid, messageId);
+    this.type = transcriptionType;
+    this.chatMutex = new Mutex();
+  }
+
+  override async startPublishingLoop() {
+    this.isRunning = true;
+    let sleepInterval = this.computeSleepInterval();
+    let isComplete = false;
+    while (this.isRunning && !isComplete) {
+      this.currentPlayoutTime += sleepInterval;
+      this.currentCharacterIndex = Math.floor(this.currentPlayoutTime * this.charsPerSecond);
+      isComplete = this.textIsComplete && this.currentCharacterIndex >= this.currentText.length;
+      switch (this.type) {
+        case TranscriptionType.CHAT:
+        case TranscriptionType.CHAT_AND_TRANSCRIPTION:
+          await this.sendChatMessage();
+        case TranscriptionType.TRANSCRIPTION:
+        case TranscriptionType.CHAT_AND_TRANSCRIPTION:
+          await this.publishTranscription(false);
+        default:
```

Comment:
this switch looks wrong to me - won't chat only still fall through to transcription?  a switch doesn't seem like the right conditional type to use for a-b-or-both behavior